compiler/hash-parser/src/token.rs

```diff
@@ -183,10 +306,15 @@ pub enum TokenAtom {
 
     Keyword(Keyword),
 
-    /// Delimiter: '(' '{', '[' and right hand-side variants, useful for error reporting and messages
+    /// General classification of an identifier
+    // @@Cleanup: find a better way to describe when we expect an identifier
+    GenericIdent,
```

Comment:
Why is this called GenericIdent?

Reply:
It's meant to represent a Identifier that starts with a capitol letter.